vendor/github.com/docker/go-plugins-helpers/volume/api.go

```diff
@@ -120,3 +134,27 @@ func (h *Handler) handle(name string, actionCall actionHandler) {
 		sdk.EncodeResponse(w, res, res.Err)
 	})
 }
+
+func (h *Handler) handleMount(name string, actionCall mountActionHandler) {
+	h.HandleFunc(name, func(w http.ResponseWriter, r *http.Request) {
+		var req MountRequest
+		if err := sdk.DecodeRequest(w, r, &req); err != nil {
+			return
+		}
+
+		res := actionCall(req)
+		sdk.EncodeResponse(w, res, res.Err)
+	})
+}
+
+func (h *Handler) handleUnmount(name string, actionCall unmountActionHandler) {
+	h.HandleFunc(name, func(w http.ResponseWriter, r *http.Request) {
+		var req UnmountRequest
+		if err := sdk.DecodeRequest(w, r, &req); err != nil {
```

Comment:
These are external code, I'm just sync'ing the go-plugin-helpers in our
repo.

On Tue, Aug 2, 2016 at 10:28 PM, Andrew J. Stone <notifications@github.com>
wrote:

> In vendor/github.com/docker/go-plugins-helpers/volume/api.go
> <https://github.com/vmware/docker-volume-vsphere/pull/566#discussion_r73194448>
> :
>
> > +func (h *Handler) handleMount(name string, actionCall mountActionHandler) {
> > +	h.HandleFunc(name, func(w http.ResponseWriter, r *http.Request) {
> > +		var req MountRequest
> > +		if err := sdk.DecodeRequest(w, r, &req); err != nil {
> > +			return
> > +		}
> > +
> > +		res := actionCall(req)
> > +		sdk.EncodeResponse(w, res, res.Err)
> > +	})
> > +}
> > +
> > +func (h *Handler) handleUnmount(name string, actionCall unmountActionHandler) {
> > +	h.HandleFunc(name, func(w http.ResponseWriter, r *http.Request) {
> > +		var req UnmountRequest
> > +		if err := sdk.DecodeRequest(w, r, &req); err != nil {
>
> Same as in handleMount. Don't swallow the error.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vmware/docker-volume-vsphere/pull/566/files/b2821e15e2c4f133635a3778e39728546056c175#r73194448>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/APHseG3utmrDyGUa2nt5xsHv8Dgm7ZRvks5qb3cfgaJpZM4JaPm_>
> .
>